core/storage/btree.rs

```diff
@@ -1248,7 +1294,9 @@ impl BTreeCursor {
         if record_buf.len() <= max_local {
             // enough allowed space to fit inside a btree page
             cell_payload.extend_from_slice(record_buf.as_slice());
-            cell_payload.resize(cell_payload.len() + 4, 0);
+            if let InsertType::RegularRecord = insert_type {
+                cell_payload.resize(cell_payload.len() + 4, 0);
+            }
```

Comment:
this is probably wrong too, mainly because ALL tables are the same, even though there is a schema table which is used internally. Cells have a format and they should stick to it. In this case I could've done better documenting this padding XD but if you want to add a comment to explain it and most likely leave it as it was. For more information you can check the format here https://www.sqlite.org/fileformat.html.